enhancements/baremetal/add-bare-metal-host-to-vsphere.md

```diff
@@ -0,0 +1,386 @@
+- Autoscaling or machine/cluster API management of nodes
+- Bare metal API integration
+
+## Proposal
+
+Nodes added to a platform vSphere cluster are expected to be initialized by the cloud controller manager. Otherwise, the nodes will remain tainted and will not join the cluster. To ensure the bare metal nodes are initialized, they are ignited with `platform: none`. Once the bare metal nodes join the cluster, the CCM recognizes the node is there but not a part of vCenter. While this results in warnings being logged by the CCM, no events are generated and the cloud controller manager operator remains available. 
+
+The CSI operator will attempt to schedule daemonset pods on all nodes and [tolerates most taints](https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/4479e2418f38cb93b5da4df7e043aff71a20cccc/manifests/vanilla/vsphere-csi-driver.yaml#L565-L569) due to its nodeSelector of any linux node. This results in the cluster storage operator progressing indefinitely since the CSI driver schedules a csi pod onto the bare metal node that crash loops indefinitely. The CSI operator will need to be modified to configure the node daemonset to only target vSphere nodes.
+
+Another issue the CSI operator has with the non vSphere node is that it fails to find the node's VM in vSphere.  This is expected since its not a vSphere node.  We will need to update the vmware csi driver operator's `checkNode` logic to ignore any node that is not a vSphere node.
+
+### Workflow Description
+
+#### Adding a New Node Without Cloud Provider Integration to a Platform vSphere cluster
+
+1. Install a `platform: vsphere` cluster
+2. Download the RHCOS Live CD which aligns with the installed version of OpenShift.
+3. Obtain or create a worker.ign file. This will be used to bootstrap the bare metal node.
+4. Boot the new bare metal host from the RHCOS Live CD.
+5. Install RHCOS:
+```bash=
+coreos-installer install /dev/sdX --insecure-ignition --ignition-url=https://path-to-compute-ignition --platform=metal
+```
+6. Reboot the node.
+7. Approve CSRs for the node
+
+Note: The [vSphere CSI driver daemonset](https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/4479e2418f38cb93b5da4df7e043aff71a20cccc/manifests/vanilla/vsphere-csi-driver.yaml#L565-L569) tolerates all taints. I was able to disable it by making the operator unmanaged and removing the tolerations.
+
+### API Extensions
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+Does not have an impact.
+
+#### Standalone Clusters
+
+#### Single-node Deployments or MicroShift
+
+Does not have an impact.
+
+### Implementation Details/Notes/Constraints
+
+#### VMware CSI Driver Operator
+
+The VMware CSI Driver operator is being enhanced in the following ways:
+1. Modify vmware-vsphere-csi-driver-node daemonset to only get deployed onto vsphere linux nodes
+2. Enhance `checkOnNode` to ignore nodes that are not flagged as being a vSphere node.
+
```

Comment:
[vsphere problem detector](https://github.com/openshift/vsphere-problem-detector/blob/main/pkg/operator/operator.go) was sensitive to other intel nodes as it could not get the UUID on the call back. You may want to consider a section on this.